File: tests/workflow/producer/rust/src/producer/events/connected.rs
use super::{identification, producer::Control, scope::Scope, Context};
use crate::stat::Alias;
use clibri::server;

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E>>(
    scope: &mut Scope<'_, E, C>,
) -> Result<(), String> {
    scope.context.add_stat(scope.identification.uuid());
    scope
        .context
        .inc_stat(scope.identification.uuid(), Alias::Connected);
    Ok(())
}

File: tests/workflow/producer/typescript/src/producer/events/connected.ts
import {
	Identification,
	Filter,
	Producer,
	Context,
	Protocol,
} from "../implementation/events";
import { Alias } from "../../stat";

export function emit(
	consumer: Identification,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<void> {
	context.connected(consumer.uuid());
	const stat = context.getStat(consumer.uuid());
	stat.case(Alias.Connected);
	return Promise.resolve();
}
